src/Raku/ast/variable-declaration.rakumod

```diff
@@ -726,10 +731,12 @@ class RakuAST::VarDeclaration::Simple
                     }
 
                     my $value := $expression.compile-time-value;
-                    unless nqp::istype($value,$vartype) {
+                    if !nqp::istype($value,$vartype) &&
+                        # nqp::istype($vartype,Numeric) returns 0 here, so check all relevant types
+                        (nqp::istype($vartype,Int) || nqp::istype($vartype,Num) || nqp::istype($vartype,Rat) || nqp::istype($vartype,FatRat)) {
```

Comment:
Are these types actually known here?  Or will they just be `NQPMu` at runtime?

Reply:
This seems to work as expected with all variants I tried. It even fixes a bug in base where the correct exception was not thrown for `FatRat`.

Reply:
`Int` and `Num` are fine but `Rat` and `FatRat` are not known in the bootstrap and will therefore be silently mapped to NQPMu. If this code works as intended anyway, that suggests that the tests are insufficient to actually trigger this, or there's something else that's fishy. Btw. `Numeric` is also unknown to the bootstrap, which explains why the istype check always fails (it's really `nqp::istype($vartype, NQPMu)`.
If you need these types, you need to resolve them first (see `RakuAST::ImplicitLookups`).

Reply:
@niner Since I actually have the resolver object in scope and because there are a variable number of implicit lookups, I took the approach of using the resolver directly. I hope that is okay.